secure_message/repository/retriever.py

```diff
@@ -31,40 +31,29 @@ def unread_message_count(user):
         return result
 
     @staticmethod
-    def thread_count_by_survey(user, survey, label=None):
+    def thread_count_by_survey(user, survey, is_closed):
         """Count users threads for a specific survey"""
 
+        if not user.is_internal:
+            logger.info("Thread count should be internal users only", user_uuid=g.user.user_uuid)
+            abort(403)
```

Comment:
The checking of permission and abort should be done in the `ThreadCounter` resource. The retriever should just be concerned with getting data from the database and returning it.